eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -88,14 +88,42 @@ case class Commitments(channelVersion: ChannelVersion,
     // we need to base the next current commitment on the last sig we sent, even if we didn't yet receive their revocation
     val remoteCommit1 = remoteNextCommitInfo.left.toOption.map(_.nextRemoteCommit).getOrElse(remoteCommit)
     val reduced = CommitmentSpec.reduce(remoteCommit1.spec, remoteChanges.acked, localChanges.proposed)
-    val feesMsat = if (localParams.isFunder) commitTxFee(remoteParams.dustLimit, reduced).toMilliSatoshi else 0.msat
-    (reduced.toRemote - remoteParams.channelReserve.toMilliSatoshi - feesMsat).max(0 msat)
+    val balanceNoFees = (reduced.toRemote - remoteParams.channelReserve).max(0 msat)
+    if (localParams.isFunder) {
+      // The funder always pays the on-chain fees, so we must subtract that from the amount we can send.
+      val commitFees = commitTxFee(remoteParams.dustLimit, reduced).toMilliSatoshi
+      val htlcFees = htlcOutputFee(reduced.feeratePerKw)
+      if (balanceNoFees - commitFees < offeredHtlcTrimThreshold(remoteParams.dustLimit, reduced)) {
```

Comment:
Why "offered" htlc?
`availableBalanceForSend` considers sending an HTLC. The commitment tx considered is the remote's. Hence it is a "received" HTLC.
`availableBalanceForReceive` considers receiving an HTLC. The commitment tx considered is the local's. Hence it is a "received" HTLC.
i.e. I believe both methods should be considering "received" HTLCs (in terms of on-chain outputs).